src/Sulu/Component/Rest/ListBuilder/Metadata/General/Driver/XmlDriver.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Rest\ListBuilder\Metadata\General\Driver;
+
+use Metadata\Driver\AbstractFileDriver;
+use Metadata\Driver\DriverInterface;
+use Metadata\MergeableClassMetadata;
+use Sulu\Component\Rest\ListBuilder\Metadata\General\PropertyMetadata;
+use Sulu\Component\Util\XmlUtil;
+use Symfony\Component\Config\Util\XmlUtils;
+
+/**
+ * Parses data from xml and returns doctrine-metadata.
```

Comment:
I have seen this `doctrine-metadata` already somewhere else, and it is a bit confusing IMO. Sounds like there are the `*.orm.xml` files from the doctrine ORM are generated here. Maybe write `field descriptor metadata for doctrine entities`?